### PR TITLE
1.3.0-beta

### PR DESCRIPTION
**Pull request by imtoori**
LLC
- Added support for pinned messages in offline storage
- Update userAgent header to specify the platform
- Minor fixes
- QueryChannels now returns a Stream and fetches the channels from storage before calling the api

UI
- Added MessageInputTheme
- Fixed overflow in MessageInput animation
- Delete only image on image gallery
- Close keyboard when sending a command
- Exposed customAttachmentBuilders through MessageListView

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart [hunk 1]
```diff
@@ -27,7 +27,9 @@ class AttachmentUploadStateBuilder extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    if (attachment.uploadState == null) return Offstage();
+    if (message.status == null || message.status == MessageSendingStatus.sent) {
+      return Offstage();
```
Comment (Nash0x7E2) [inline packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart:31]: Do you need an `Offstage` here? Would a `SizedBox` be better for a placeholder widget?

File: packages/stream_chat_flutter/lib/src/stream_chat_theme.dart [hunk 2]
```diff
@@ -873,6 +845,65 @@ class ChannelHeaderTheme {
   }
 }
 
+class MessageInputTheme {
```
Comment (Nash0x7E2) [inline packages/stream_chat_flutter/lib/src/stream_chat_theme.dart:848]: Please add documentation above the class and constructor.

**Review by Nash0x7E2 — COMMENTED** [review]
Nice job, left two minor nitpicks.

**Comment by Nash0x7E2** [comment]
Looks like we are missing documentation in a few places:

```shell 
[stream_chat]: 
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:5:6 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:6:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:7:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:8:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:9:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:10:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:11:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:12:3 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:15:7 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:17:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:18:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:19:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:20:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:21:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:22:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:23:19 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:25:21 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector.dart:46:27 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector_io.dart:4:18 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector_stub.dart:3:18 - (public_member_api_docs)
[stream_chat]:   info - Document all public members at lib/src/platform_detector/platform_detector_web.dart:3:18 - (public_member_api_docs)
[stream_chat]: 
```

File: packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart (hunk 1 above)
Comment (imtoori) [reply packages/stream_chat_flutter/lib/src/attachment/attachment_upload_state_builder.dart:31]: I don't think it makes much difference
We use it also in other places

**Review by Nash0x7E2 — APPROVED** [review]
LGTM

**Pull request merged**